src/main/java/duke/task/TaskArray.java

```diff
@@ -0,0 +1,367 @@
+import java.nio.file.Paths;
+
+import java.util.Scanner;
+
+public class TaskArray {
+    public static final int MAX_SIZE = 100;
+    public static ArrayList<Task> taskArrayList = new ArrayList<>();
+
+    public static String DELIMITER_EMPTY_STRING = "";
+    public static String DELIMITER_SLASH = "/";
+    public static String DELIMITER_SEMI_COLON = ":";
+    public static String DELIMITER_CHARACTER = " ";
+    public static String DIVIDER = " | ";
+
+    public static String DIRECTORY = "data";
+    public static String FILEPATH = "data/data.txt";
+
+    public static Task returnTaskFromInput(String input, String keyword) throws
+            IllegalEmptyDescriptionException, IllegalPrepositionWithoutDate {
+        Task currentTask;
+        switch (keyword) {
+        case "event":
+            String[] eventDetails = parseString(keyword, input);
+            currentTask = new Event(eventDetails[0], eventDetails[1]);
+            break;
+        case "deadline":
+            String[] deadlineDetails = parseString(keyword, input);
+            currentTask = new Deadline(deadlineDetails[0], deadlineDetails[1]);
+            break;
+        default:
+            String task = returnTaskDescription(input, keyword);
+            currentTask = new Todo(task);
+            break;
+        }
+        return currentTask;
+
+    }
+
+    /**
+     * Returns a task object that has been instantiated according to its type of task
+     *
+     * @param input   Given by the user
+     * @param keyword The type of the task
+     * @return A task object
+     * @throws IllegalEmptyDescriptionException If the user does not enter a valid description
+     * @throws IllegalPrepositionWithoutDate    If the user does not follow the
+     *                                          required format for deadline and events
+     */
+
+
+    public static void readTask(String input, String keyword, boolean writeToFile) {
+
+        try {
+            PrintMethod.printLines();
+            Task currentTask = returnTaskFromInput(input, keyword);
+            addToTasks(currentTask, writeToFile);
+
+        } catch (IllegalEmptyDescriptionException i) {
+            PrintMethod.printEmptyDescription(keyword);
+
+        } catch (ArrayIndexOutOfBoundsException | IllegalPrepositionWithoutDate a) {
+            PrintMethod.printEmptyDate(keyword);
+        } catch (IOException e) {
+            System.out.println("Something went wrong" + e.getMessage());
+        }
+        PrintMethod.printLines();
+    }
+
+
+    public static String returnTaskDescription(String command, String typeOfTask) throws
+            IllegalEmptyDescriptionException {
+        String task = command.replace(typeOfTask, DELIMITER_EMPTY_STRING).strip();
+        if (task.equals("")) {
+            throw new IllegalEmptyDescriptionException();
+        }
+        return task;
+    }
+
+    public static void readDoneAndDelete(String command, String keyword, boolean writeFile) {
```

Comment:
The name of a boolean variable should be in the format of "isX", for example, "isWriteFileDone". Thus, it is better if the bollean argument "writeFile" can follow this format.